core/foundation/test/TypeTraits.cxx

```diff
@@ -0,0 +1,143 @@
+#include "ROOT/TypeTraits.hxx"
```

Comment:
We were using EntityToTest`Tests.cxx` pattern. Having something/test/test*.cxx looks very weird...